### PR TITLE
Implemented compat for the get-the-image plugin

### LINKED ISSUE #381: Image in post not displaying
# Steps required to reproduce the problem

1. 
2. 
3. 

# Expected Result

*Pulling image from the article and publish in instant article.

# Actual Result

* I'm getting an instant article without image even though i input image in the article.
Though, image will pick up only when input in feature attachment

---

** Version Info **

* Plugin version: 3.0.1
* WordPress version:  wordpress 4.5.3
* PHP version:  5.5

### PR DESCRIPTION
**Pull request by everton-rosario**
This PR:

* [x] Adds configuration rules for the get-the-image plugin
* [x] Compat layer setup with the new plugin

Fixes #381

### CODE REVIEW

File: compat.php [hunk 1]
```diff
@@ -7,6 +7,11 @@
  * @package default
  */
 
+if ( !function_exists( 'is_plugin_active' ) ) {
```
Comment (diegoquinteiro) [inline compat.php]: Why do we need this now?

File: compat/get-the-image-rules-configuration.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "rules": [{
+      "class": "ImageInsideParagraphRule",
+      "selector": "figure.wp-caption",
```
Comment (diegoquinteiro) [inline compat/get-the-image-rules-configuration.json]: Did you test to confirm the caption is being read too?

File: compat.php (hunk 1 above)
Comment (everton-rosario) [inline compat.php]: This should be how we check if any plugin is active.

File: compat/get-the-image-rules-configuration.json (hunk 2 above)
Comment (everton-rosario) [inline compat/get-the-image-rules-configuration.json]: Yes.

**Comment by adavize4u** [comment]
Still having little problem:
1. Sometimes, text striped from the transform post, making our facebook update useless.
2. Image mostly appear when article published with new image, but old image picked from the site uploads, it sometimes not displaying image.

**Comment by everton-rosario** [comment]
@adavize4u Can you share the markup sent to transformer, for the cases you've mentioned?

To get the transformation code, just click in the link "**[Toggle debug information]**" in the "Instant Articles" meta-box and then copy and past the both editor's text: "**Source Markup**" and "**Transformed Markup**"

This way I can reproduce and check where are the corner cases.

Thanks, @adavize4u

**Comment by everton-rosario** [comment]
Hey @philipjohn/@diegoquinteiro, why this PR didn't get thru to the release 3.1?

**Comment by philipjohn** [comment]
Playing Milestone tennis :)

**Pull request merged**